src/screens/titlescreen/LoginComponent.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState, MouseEvent, ChangeEvent } from "react";
+import Network from "../../network/Network";
+
+type Props = {
+  setLoggedIn: (loggedIn: boolean) => void;
+  loggedIn: boolean;
+  setTitleMenuMode: (titleMenuMode: string) => void;
+}
+
+export default function LoginComponent ({ setLoggedIn, setTitleMenuMode, loggedIn } : Props): JSX.Element {
+
+  const [resultView, setResultView] = useState(<div>Please enter your username and token above to continue.</div>);
+  const [inputUsername, setInputUsername] = useState('');
+  const [inputToken, setInputToken] = useState('');
+
+  const network = new Network();
+
+  const displayErrorMessage = (errorMsg: string) => {
+    setResultView(<div>Error: {errorMsg}</div>);
+  };
+
+  const handleChange = (event: ChangeEvent) => {
+    event.preventDefault();
+    switch (event.target.id) {
+      case "login-name":
+        setInputUsername((event.target as HTMLInputElement).value);
+        console.log("Writing to loginName.");
+        break;
+      case "token":
+        setInputToken((event.target as HTMLInputElement).value);
+        console.log("Writing to token.");
+        break;
+      default:
+        console.log("Default change handler reached.");
+    }
+  };
+
+  const handleSubmit = (event: MouseEvent) => {
+    event.preventDefault();
+    console.log("Logging in.");
+    network.authenticateUser(
+      inputUsername,
+      inputToken,
+      (errorMessage: string) => {
+        if (errorMessage) {
+          displayErrorMessage(errorMessage);
+        } else {
+          setLoggedIn(true);
+          setResultView(
+            <div>
+              <div>Successfully logged in as {inputUsername}!</div>
+            </div>,
+          );
+        }
+      },
+    );
+  };
+
+  const handleBack = (event: MouseEvent) => {
+    event.preventDefault;
```

Comment:
I think this should be event.preventDefault();

Reply:
Excellent point.